gen2-mjpeg-streaming/main.py

```diff
@@ -75,85 +75,95 @@ class ThreadedHTTPServer(ThreadingMixIn, HTTPServer):
-
-colorCam.preview.link(spatialDetectionNetwork.input)
-if syncNN:
-    spatialDetectionNetwork.passthrough.link(xoutRgb.input)
-else:
-    colorCam.preview.link(xoutRgb.input)
-
-spatialDetectionNetwork.out.link(xoutNN.input)
-spatialDetectionNetwork.boundingBoxMapping.link(xoutBoundingBoxDepthMapping.input)
-
-stereo.depth.link(spatialDetectionNetwork.inputDepth)
-spatialDetectionNetwork.passthroughDepth.link(xoutDepth.input)
+def create_pipeline(depth):
+    # Start defining a pipeline
+    pipeline = dai.Pipeline()
+    pipeline.setOpenVINOVersion(version=dai.OpenVINO.Version.VERSION_2021_2)
+    # Define a source - color camera
+    colorCam = pipeline.createColorCamera()
+    if depth:
+        mobilenet = pipeline.createMobileNetSpatialDetectionNetwork()
+        monoLeft = pipeline.createMonoCamera()
+        monoRight = pipeline.createMonoCamera()
+        stereo = pipeline.createStereoDepth()
+    else:
+        mobilenet = pipeline.createMobileNetDetectionNetwork()
+
+    colorCam.setPreviewSize(300, 300)
+    colorCam.setResolution(dai.ColorCameraProperties.SensorResolution.THE_1080_P)
+    colorCam.setInterleaved(False)
+    colorCam.setColorOrder(dai.ColorCameraProperties.ColorOrder.BGR)
+
+    mobilenet.setBlobPath(nnBlobPath)
+    mobilenet.setConfidenceThreshold(0.5)
+    mobilenet.input.setBlocking(False)
+
+    if depth:
+        monoLeft.setResolution(dai.MonoCameraProperties.SensorResolution.THE_400_P)
+        monoLeft.setBoardSocket(dai.CameraBoardSocket.LEFT)
+        monoRight.setResolution(dai.MonoCameraProperties.SensorResolution.THE_400_P)
+        monoRight.setBoardSocket(dai.CameraBoardSocket.RIGHT)
+
+        # Setting node configs
+        stereo.setOutputDepth(True)
+        stereo.setConfidenceThreshold(255)
+        stereo.depth.link(mobilenet.inputDepth)
+
+        mobilenet.setBoundingBoxScaleFactor(0.5)
+        mobilenet.setDepthLowerThreshold(100)
+        mobilenet.setDepthUpperThreshold(5000)
+
+        monoLeft.out.link(stereo.left)
+        monoRight.out.link(stereo.right)
+
+    xoutRgb = pipeline.createXLinkOut()
+    xoutRgb.setStreamName("rgb")
+    colorCam.preview.link(mobilenet.input)
+    if syncNN:
+        mobilenet.passthrough.link(xoutRgb.input)
+    else:
+        colorCam.preview.link(xoutRgb.input)
+
+
+    xoutNN = pipeline.createXLinkOut()
+    xoutNN.setStreamName("detections")
+    mobilenet.out.link(xoutNN.input)
+
+    if depth:
+        xoutBoundingBoxDepthMapping = pipeline.createXLinkOut()
+        xoutBoundingBoxDepthMapping.setStreamName("boundingBoxDepthMapping")
+        mobilenet.boundingBoxMapping.link(xoutBoundingBoxDepthMapping.input)
+
+        xoutDepth = pipeline.createXLinkOut()
+        xoutDepth.setStreamName("depth")
+        mobilenet.passthroughDepth.link(xoutDepth.input)
+
+    return pipeline
 
 # Pipeline is defined, now we can connect to the device
-with dai.Device(pipeline) as device:
+with dai.Device(dai.OpenVINO.Version.VERSION_2021_2) as device:
```

Comment:
Can we use blobconverter to fetch the blob with latest OpenVINO version?

Reply:
done, thanks for suggestion:)